cypress/integration/dimensions/period.cy.js

```diff
@@ -0,0 +1,484 @@
+    unselectAllItemsByButton,
+    switchToFixedPeriods,
+    expectSelectablePeriodsAmountToBeLeast,
+    openFixedPeriodsTypeSelect,
+    switchFixedPeriodType,
+    expectRelativePeriodTypeSelectToNotContain,
+    expectFixedPeriodTypeSelectToNotContain,
+    expectFixedPeriodTypeToBe,
+} from '../../elements/dimensionModal'
+import { openDimension } from '../../elements/dimensionsPanel'
+import { goToStartPage } from '../../elements/startScreen'
+
+const defaultRelativePeriod = 'Last 3 months'
+const systemSettingsBody = {
+    keyAnalysisRelativePeriod: 'LAST_3_MONTHS',
+    keyAnalysisDigitGroupSeparator: 'COMMA',
+    keyHideDailyPeriods: false,
+    keyHideWeeklyPeriods: false,
+    // TODO: Add Bi-weekly periods once it's supported
+    keyHideMonthlyPeriods: false,
+    keyHideBiMonthlyPeriods: false,
+    keyIgnoreAnalyticsApprovalYearThreshold: -1,
+    keyDateFormat: 'yyyy-MM-dd',
+}
+
+describe('Period dimension', () => {
+    describe('initial state', () => {
+        it('navigates to the start page', () => {
+            cy.intercept(
+                /systemSettings(\S)*keyAnalysisRelativePeriod(\S)*/,
+                req => {
+                    req.reply(res => {
+                        res.send({ body: systemSettingsBody })
+                    })
+                }
+            )
+            goToStartPage()
+        })
+        it('opens the period dimension modal', () => {
+            openDimension(DIMENSION_ID_PERIOD)
+            expectPeriodDimensionModalToBeVisible()
+        })
+        it('modal has a title', () => {
+            expectDimensionModalToContain('Period')
+        })
+        it('1 period is selected', () => {
+            expectSelectedPeriodsAmountToBe(1)
+        })
+        it(`period '${defaultRelativePeriod}' is selected`, () => {
+            expectItemToBeSelected(defaultRelativePeriod)
+        })
+        it("period category is 'Relative'", () => {
+            expectRelativeToBeSelected()
+        })
+        it("period type is 'Months'", () => {
+            expectRelativePeriodTypeToBe('Months')
+        })
+        it('a period can be unselected by double click', () => {
+            unselectItemByDoubleClick(defaultRelativePeriod)
+            expectNoPeriodsToBeSelected()
+        })
+        it('a period can be selected by double click', () => {
+            selectItemByDoubleClick(defaultRelativePeriod)
+            expectItemToBeSelected(defaultRelativePeriod)
+        })
+        it('a period can be unselected by button', () => {
+            unselectItemByButton(defaultRelativePeriod)
+            expectNoPeriodsToBeSelected()
+        })
+        it('a period can be selected by button', () => {
+            selectItemByButton(defaultRelativePeriod)
+            expectItemToBeSelected(defaultRelativePeriod)
+        })
+        it('all can be unselected by button', () => {
+            unselectAllItemsByButton()
+        })
+    })
+    describe('relatives period', () => {
```

Comment:
Wondering if this describe and the next one on line 124 should be a subset of the `initial state` describe above since the test is not re-visiting the start page? Kind of encapsulates the default state where there are no system settings restrictions.

Reply:
Good idea, done!